frontend/src/utils/patterns.ts

```diff
@@ -0,0 +1,9 @@
+export const emailPattern = {
```

Comment:
These are validation rules.  Why not keep them in validation.ts?  
Or maybe rename this to "validatePatterns"
I expect to see only the patterns in a file named "patterns.ts"

Reply:
Moved to `validation.ts` file.